ehsm_kms_service/quote_manager_apis.js

```diff
@@ -0,0 +1,270 @@
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * The UploadQuotePolicy Support uploading MRenclave and MRsigner and returning new policyid.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}mr_enclave(String [1~1024]) : A string of length 64. eg. '870c42c59bc74c7ad22869411709e4f78ac3c76add6693bb43296b03362e5038'
+ *          ==> {r}mr_signer(String [1~1024]) : A string of length 64. eg. 'c30446b4be9baf0f69728423ea613ef81a63e72acf7439fa0549001fd5482835'
+ * @returns {Object}
+ *          ==> {r}policyId(String) : a new policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ */
+const uploadQuotePolicy = async (res, appid, payload, DB) => {
+    try {
+        const { mr_enclave, mr_signer } = payload
+        if (!checkStringParam(mr_enclave, true)) {
+            res.send(_result(400, `mr_enclave cannot be empty, must be string.`))
+            return
+        }
+        if (!checkStringParam(mr_signer, true)) {
+            res.send(_result(400, `mr_signer cannot be empty, must be string.`))
+            return
+        }
+        if(mr_enclave.length != MR_ENCLAVE_LENGTH){
+            res.send(_result(400, `mr_enclave length must be ${MR_ENCLAVE_LENGTH}.`))
+            return
+        }
+        if(mr_signer.length != MR_SIGNER_LENGTH){
+            res.send(_result(400, `mr_enclave length must be ${MR_ENCLAVE_LENGTH}.`))
+            return
+        }
+        let policyId = uuidv4();
+        const createTime = new Date().getTime()
+        await DB.insert({
+            _id: `quote_policy:${policyId}`,
+            appid,
+            policyId,
+            mr_enclave,
+            mr_signer,
+            createTime
+        })
+        let result = {
+            policyId
+        }
+        res.send(_result(200, 'Upload quote policy success.', result))
+        return
+    } catch (e) {
+        console.info('uploadQuotePolicy :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * query a quote policy information by policyid.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}policyId(String) : a policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ * @returns {Object}
+ *          ==> {r}policyId(String) : a policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ *          ==> {r}mr_enclave(String) : A string of length 64. eg. '870c42c59bc74c7ad22869411709e4f78ac3c76add6693bb43296b03362e5038'
+ *          ==> {r}mr_signer(String) : A string of length 64. eg. 'c30446b4be9baf0f69728423ea613ef81a63e72acf7439fa0549001fd5482835'
+ */
+const getQuotePolicy = async (res, appid, payload, DB) => {
+    try {
+        const { policyId } = payload
```

Comment:
do these 2 new APIs will verify the HAMC sign of the parameter and can they provide the isolation with the APPID?

Reply:
yes